pkg/zones/wellknown.go

```diff
@@ -386,10 +258,49 @@ func GuessCloudForZone(zone string) (kops.CloudProviderID, bool) {
 	return matches[0], true
 }
 
-func WellKnownZonesForCloud(matchCloud kops.CloudProviderID) []string {
+func WellKnownZonesForCloud(matchCloud kops.CloudProviderID, prefix string) []string {
 	switch matchCloud {
 	case kops.CloudProviderAWS:
-		return awsZones
+		prefix = strings.ToLower(prefix)
+		var found []string
+
+		for _, partition := range endpoints.DefaultResolver().(endpoints.EnumPartitions).Partitions() {
+			for _, region := range partition.Regions() {
+				regionName := strings.ToLower(region.ID())
+				if !strings.HasPrefix(regionName, prefix) {
+					// For the guess-cloud case, if the zone name is in the form of zones belonging
+					// to a region, that's good enough.
+					if len(prefix) == len(regionName)+1 && strings.HasPrefix(prefix, regionName) {
+						found = append(found, prefix)
+					}
+					if !strings.HasPrefix(prefix, regionName+"-") {
```

Comment:
What does `regionName+"-"` represent?

Reply:
That's the prefix for Local Zones and Wavelength Zones. Basically, any zone suffix that isn't a single letter.

Reply:
Added comment as requested in Office Hours.

Reply:
@hakman is this good to go now?